NXP: Shield rk055hdmipi4ma0 fails to build with mimxrt595_evk_cm33

With v3.6.0-rc3 tag, the [display sample](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/drivers/display) fails to build for the board `mimxrt595_evk_cm33` using the display shield `rk055hdmipi4ma0`.  

Reverting to 638e3d5d9c40b27ef745d6ebb131406aad728f53 when this shield was introduced, the sample builds and runs well on this hardware.

Below is the build log.

```
$ west build -b mimxrt595_evk_cm33 -d ../../display_RT500/ samples/drivers/display/ --pristine -- -DSHIELD=rk055hdmipi4ma0
-- west build: making build dir /home/snellda/zNXPmicro/display_RT500 pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/snellda/zNXPmicro/zephyrproject/zephyr/samples/drivers/display
-- CMake version: 3.28.3
-- Found Python3: /home/snellda/venv/bin/python (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/snellda/.cache/zephyr
-- Zephyr version: 3.6.0-rc3 (/home/snellda/zNXPmicro/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: mimxrt595_evk_cm33
-- Shield(s): rk055hdmipi4ma0
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.4 (/home/snellda/zephyr-sdk-0.16.4)
-- Found toolchain: zephyr 0.16.4 (/home/snellda/zephyr-sdk-0.16.4)
-- Found Dtc: /home/snellda/zephyr-sdk-0.16.4/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/snellda/zNXPmicro/zephyrproject/zephyr/boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.dts
-- Found devicetree overlay: /home/snellda/zNXPmicro/zephyrproject/zephyr/boards/shields/rk055hdmipi4ma0/rk055hdmipi4ma0.overlay
-- Found devicetree overlay: /home/snellda/zNXPmicro/zephyrproject/zephyr/boards/shields/rk055hdmipi4ma0/boards/mimxrt595_evk_cm33.overlay
devicetree error: /home/snellda/zNXPmicro/zephyrproject/zephyr/boards/shields/rk055hdmipi4ma0/boards/mimxrt595_evk_cm33.overlay:16 (column 1): parse error: undefined node label 'touch_controller_rk055hdmipi4ma0'
-- In: /home/snellda/zNXPmicro/display_RT500/zephyr, command: /home/snellda/venv/bin/python;/home/snellda/zNXPmicro/zephyrproject/zephyr/scripts/dts/gen_defines.py;--dts;/home/snellda/zNXPmicro/display_RT500/zephyr/zephyr.dts.pre;--dtc-flags;'-Wno-simple_bus_reg';--bindings-dirs;/home/snellda/zNXPmicro/zephyrproject/zephyr/dts/bindings;--header-out;/home/snellda/zNXPmicro/display_RT500/zephyr/include/generated/devicetree_generated.h.new;--dts-out;/home/snellda/zNXPmicro/display_RT500/zephyr/zephyr.dts.new;--edt-pickle-out;/home/snellda/zNXPmicro/display_RT500/zephyr/edt.pickle;--vendor-prefixes;/home/snellda/zNXPmicro/zephyrproject/zephyr/dts/bindings/vendor-prefixes.txt
CMake Error at /home/snellda/zNXPmicro/zephyrproject/zephyr/cmake/modules/dts.cmake:268 (message):
  gen_defines.py failed with return code: 1
Call Stack (most recent call first):
  /home/snellda/zNXPmicro/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:129 (include)
  /home/snellda/zNXPmicro/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/snellda/zNXPmicro/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -DWEST_PYTHON=/home/snellda/venv/bin/python -B/home/snellda/zNXPmicro/display_RT500 -GNinja -DBOARD=mimxrt595_evk_cm33 -DSHIELD=rk055hdmipi4ma0 -S/home/snellda/zNXPmicro/zephyrproject/zephyr/samples/drivers/display
```

--- a/boards/shields/rk055hdmipi4ma0/boards/mimxrt595_evk_mimxrt595s_cm33.overlay
+++ b/boards/shields/rk055hdmipi4ma0/boards/mimxrt595_evk_mimxrt595s_cm33.overlay
@@ -13,7 +13,7 @@
 };
 
 /* GT911 IRQ GPIO is active low on this board, and needs probing mode */
-&touch_controller_rk055hdmipi4ma0 {
+&gt911_rk055hdmipi4ma0 {
 	irq-gpios = <&nxp_mipi_connector 29 GPIO_ACTIVE_LOW>;
 	alt-addr = <0x14>;
 };

--- a/drivers/display/display_hx8394.c
+++ b/drivers/display/display_hx8394.c
@@ -406,6 +406,37 @@ const uint8_t hx8394_cmd3[] = {0xC6U, 0xEDU};
 
 const uint8_t tear_config[] = {HX8394_SET_TEAR, HX8394_TEAR_VBLANK};
 
+static ssize_t hx8394_mipi_tx(const struct device *mipi_dev, uint8_t channel,
+			      const void *buf, size_t len)
+{
+	/* Send MIPI transfers using low power mode */
+	struct mipi_dsi_msg msg = {
+		.tx_buf = buf,
+		.tx_len = len,
+		.flags = MIPI_DSI_MSG_USE_LPM,
+	};
+
+	switch (len) {
+	case 0U:
+		msg.type = MIPI_DSI_GENERIC_SHORT_WRITE_0_PARAM;
+		break;
+
+	case 1U:
+		msg.type = MIPI_DSI_GENERIC_SHORT_WRITE_1_PARAM;
+		break;
+
+	case 2U:
+		msg.type = MIPI_DSI_GENERIC_SHORT_WRITE_2_PARAM;
+		break;
+
+	default:
+		msg.type = MIPI_DSI_GENERIC_LONG_WRITE;
+		break;
+	}
+
+	return mipi_dsi_transfer(mipi_dev, channel, &msg);
+}
+
 static int hx8394_write(const struct device *dev, const uint16_t x,
 			 const uint16_t y,
 			 const struct display_buffer_descriptor *desc,
@@ -477,7 +508,7 @@ static int hx8394_set_orientation(const struct device *dev,
 	default:
 		return -ENOTSUP;
 	}
-	return mipi_dsi_generic_write(config->mipi_dsi, config->channel, param, 2);
+	return hx8394_mipi_tx(config->mipi_dsi, config->channel, param, 2);
 }
 
 static void hx8394_get_capabilities(const struct device *dev,
@@ -557,66 +588,66 @@ static int hx8394_init(const struct device *dev)
 		k_sleep(K_MSEC(50));
 	}
 	/* Enable extended commands */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			enable_extension, sizeof(enable_extension));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     enable_extension, sizeof(enable_extension));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Set the number of lanes to DSISETUP0 parameter */
 	setmipi[1] |= (config->num_of_lanes - 1);
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-				setmipi, sizeof(setmipi));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     setmipi, sizeof(setmipi));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Set scan direction */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			address_config, sizeof(address_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     address_config, sizeof(address_config));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Set voltage and current targets */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			power_config, sizeof(power_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     power_config, sizeof(power_config));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Setup display line count and front/back porch size */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			line_config, sizeof(line_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     line_config, sizeof(line_config));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Setup display cycle counts (in counts of TCON CLK) */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			cycle_config, sizeof(cycle_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     cycle_config, sizeof(cycle_config));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Set group delay values */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			gip0_config, sizeof(gip0_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     gip0_config, sizeof(gip0_config));
 	if (ret < 0) {
 		return ret;
 	}
 
 
 	/* Set group clock selections */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			gip1_config, sizeof(gip1_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     gip1_config, sizeof(gip1_config));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Set group clock selections for GS mode */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			gip2_config, sizeof(gip2_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     gip2_config, sizeof(gip2_config));
 	if (ret < 0) {
 		return ret;
 	}
@@ -627,15 +658,15 @@ static int hx8394_init(const struct device *dev)
 	 */
 	k_msleep(1);
 	/* Set VCOM voltage config */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			vcom_config, sizeof(vcom_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     vcom_config, sizeof(vcom_config));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Set manufacturer supplied gamma values */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			gamma_config, sizeof(gamma_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     gamma_config, sizeof(gamma_config));
 	if (ret < 0) {
 		return ret;
 	}
@@ -643,15 +674,15 @@ static int hx8394_init(const struct device *dev)
 	/* This command is not documented in datasheet, but is included
 	 * in the display initialization done by MCUXpresso SDK
 	 */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			hx8394_cmd1, sizeof(hx8394_cmd1));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     hx8394_cmd1, sizeof(hx8394_cmd1));
 	if (ret < 0) {
 		return ret;
 	}
 
 	/* Set panel to BGR mode, and reverse colors */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			panel_config, sizeof(panel_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     panel_config, sizeof(panel_config));
 	if (ret < 0) {
 		return ret;
 	}
@@ -659,8 +690,8 @@ static int hx8394_init(const struct device *dev)
 	/* This command is not documented in datasheet, but is included
 	 * in the display initialization done by MCUXpresso SDK
 	 */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			hx8394_cmd2, sizeof(hx8394_cmd2));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     hx8394_cmd2, sizeof(hx8394_cmd2));
 	if (ret < 0) {
 		return ret;
 	}
@@ -668,43 +699,43 @@ static int hx8394_init(const struct device *dev)
 	/* Write values to manufacturer register banks */
 	param[0] = HX8394_SETBANK;
 	param[1] = 0x2;
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			param, 2);
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     param, 2);
 	if (ret < 0) {
 		return ret;
 	}
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			hx8394_bank2, sizeof(hx8394_bank2));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     hx8394_bank2, sizeof(hx8394_bank2));
 	if (ret < 0) {
 		return ret;
 	}
 	param[1] = 0x0;
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			param, 2);
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     param, 2);
 	if (ret < 0) {
 		return ret;
 	}
 	/* Select bank 1 */
 	param[1] = 0x1;
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			param, 2);
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     param, 2);
 	if (ret < 0) {
 		return ret;
 	}
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			hx8394_bank1, sizeof(hx8394_bank1));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     hx8394_bank1, sizeof(hx8394_bank1));
 	if (ret < 0) {
 		return ret;
 	}
 	/* Select bank 0 */
 	param[1] = 0x0;
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			param, 2);
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     param, 2);
 	if (ret < 0) {
 		return ret;
 	}
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			hx8394_bank0, sizeof(hx8394_bank0));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     hx8394_bank0, sizeof(hx8394_bank0));
 	if (ret < 0) {
 		return ret;
 	}
@@ -712,27 +743,31 @@ static int hx8394_init(const struct device *dev)
 	/* This command is not documented in datasheet, but is included
 	 * in the display initialization done by MCUXpresso SDK
 	 */
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			hx8394_cmd3, sizeof(hx8394_cmd3));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     hx8394_cmd3, sizeof(hx8394_cmd3));
 	if (ret < 0) {
 		return ret;
 	}
 
-	ret = mipi_dsi_generic_write(config->mipi_dsi, config->channel,
-			tear_config, sizeof(tear_config));
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     tear_config, sizeof(tear_config));
 	if (ret < 0) {
 		return ret;
 	}
 
-	ret = mipi_dsi_dcs_write(config->mipi_dsi, config->channel,
-				MIPI_DCS_EXIT_SLEEP_MODE, NULL, 0);
+	param[0] = MIPI_DCS_EXIT_SLEEP_MODE;
+
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     param, 1);
 	if (ret < 0) {
 		return ret;
 	}
 	/* We must delay 120ms after exiting sleep mode per datasheet */
 	k_sleep(K_MSEC(120));
-	ret = mipi_dsi_dcs_write(config->mipi_dsi, config->channel,
-				MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+
+	param[0] = MIPI_DCS_SET_DISPLAY_ON;
+	ret = hx8394_mipi_tx(config->mipi_dsi, config->channel,
+			     param, 1);
 
 	if (config->bl_gpio.port != NULL) {
 		ret = gpio_pin_configure_dt(&config->bl_gpio, GPIO_OUTPUT_ACTIVE);

